examples/erp/genseki/plugins/phone.ts

```diff
@@ -0,0 +1,77 @@
+import z from 'zod'
+
+import { phone, PhoneService, PhoneStore } from '@genseki/plugins'
+
+import { FullModelSchemas } from '../../generated/genseki/unsanitized'
+import { context, prisma } from '../helper'
+
+const body = z.object({
+  name: z.string(),
+  phone: z.string(),
+  password: z.string(),
+  email: z.email().optional(),
+})
+
+class CustomPhoneStore extends PhoneStore<typeof body> {
+  async createUser(data: z.infer<typeof body>): Promise<string> {
+    const user = await prisma.user.create({
+      data: {
+        name: data.name,
+        email: data.email,
+        phone: data.phone,
+      },
+    })
+    return user.id
+  }
+}
+
+const phoneStore = new CustomPhoneStore(prisma as any)
+
+const phoneService = new PhoneService(
+  context,
+  {
+    user: FullModelSchemas.user,
+    account: FullModelSchemas.account,
+    session: FullModelSchemas.session,
+    verification: FullModelSchemas.verification,
+  },
+  {
+    signUp: {
+      body: body,
+      onOtpSent: async () => {
+        return {
+          refCode: crypto.randomUUID().slice(0, 6),
+          token: crypto.randomUUID(),
+        }
+      },
+      onOtpVerify: async (payload) => {
+        return Math.random() < 0.5
+      },
+    },
+    changePhone: {
+      onOtpSent: async () => {
+        return {
+          refCode: crypto.randomUUID().slice(0, 6),
+          token: crypto.randomUUID(),
+        }
+      },
+      onOtpVerify: async (payload) => {
+        return Math.random() < 0.5
+      },
```

Comment:
I'm quite not recommend using random for checking otp verify even in example application 😅

Reply:
Instead, we may have mock server to mock OTP verification which is a huge task. Maybe not in this PR